### PR TITLE
all done

### PR DESCRIPTION
**Pull request by Yevhen-Pavlovskyi**
https://yevhen-pavlovskyi.github.io/js_employees_table_DOM/

### CODE REVIEW

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
1. зміг створити без salary
image
2. Також трохи не співпадає валідація коли редагуєш і коли створюєш доприкладу при створенні я зміг ввести CEO в posiiton, а редагувати на CEO вже не можу

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,373 @@
...
+
+            pushNotification(450, 190, 'Error',
+              'The name of the position must be'
+              + `contain at least ${minPositionChars} letters`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'The person\'s position has been successfully changed',
+              'success');
+            td.textContent = `${tag.value}`;
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:327]: will be ok without conversion to string

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,373 @@
...
+          if (tag.value < minSalary) {
+            td.textContent = tdText;
+
+            pushNotification(450, 190, 'Error',
+              `Salary cannot be less than ${minSalary}`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'a person\'s salary has been successfully changed', 'success');
+            td.textContent = `$${parseInt(tag.value).toLocaleString('en-EN')}`;
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:355]: you can also add `{style:"currency", currency:"USD"}` option for `toLocaleString`

Comment (ivanchuk-yurii) [reply src/scripts/main.js:355]: https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_tolocalestring_num_options3

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,373 @@
...
+    function replaceValue(tag) {
+      const emptyValue = tag.value.length === 0;
+
+      if (emptyValue) {
+        td.textContent = tdText;
+
+        pushNotification(450, 190, 'Error',
+          'The cell must not be empty!!! '
+          + 'The data are not changed!!!',
+          'error');
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:295]: closing bracket `)` should be on the same line as opening `(`, fix in other places

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,373 @@
...
+      savedText(select);
+    } else {
+      const input = document.createElement('input');
+
+      if (ageCell || salaryCell) {
+        input.type = 'number';
+      }
+
+      input.className = 'cell-input';
+      input.value = tdText;
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:265]: you should convert value for salary from string to number in right way, bc now it's not inserting previous value in the input

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,373 @@
...
+}
+
+function addForm() {
+  const form = `<form class='new-employee-form'>
+      <label>Name: <input
+          name='name'
+          type='text'
+          data-qa='name'
+          required
+        ></label>
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:86]: move text and tag on the new lines

File: src/scripts/main.js [hunk 6]
```diff
@@ -1,3 +1,373 @@
...
+
+  table.addEventListener('click', (e) => {
+    const item = e.target;
+
+    if (!item.closest('thead')) {
+      return;
+    }
+
+    sorted(item);
+    toggle *= -1;
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:29]: if you click on another header, toggle won't be cleared

File: src/scripts/main.js [hunk 7]
```diff
@@ -1,3 +1,373 @@
...
+          .sort((rowA, rowB) => toggle * (rowA.cells[column]
+            .innerHTML.localeCompare(rowB.cells[column]
+              .innerHTML)));
+
+        table.tBodies[0].append(...sortedRows);
+      } else {
+        const sortedRows = [...table.rows]
+          .slice(1, table.rows.length - 1)
+          .sort((rowA, rowB) => toggle * (+rowA.cells[column]
+            .innerHTML.split('$').join('').split(',').join('')
```
Comment (ivanchuk-yurii) [inline src/scripts/main.js:47]: use `replace`, move in helper function

**Review by ivanchuk-yurii — CHANGES_REQUESTED** [review]
try to fix changing of cell width when you click on it and input appears
sorry for extra comments((

**Pull request closed without merging**